doc/en/how-to/fastapi-integration-testing-tutorial.rst

```diff
@@ -0,0 +1,219 @@
+FastAPI Integration Testing Tutorial
+==================================
+
+This tutorial demonstrates how to write integration tests for FastAPI applications using pytest.
+
+Prerequisites
+------------
+
+* Python 3.7+
+* pytest
+* FastAPI
+* SQLAlchemy
+* pytest-asyncio
+
+Basic Setup
+----------
+
+First, let's set up the basic structure for integration testing. Create a ``tests`` directory in your project root with the following files::
+
+    your_project/
+    ├── app/
+    │   ├── main.py
+    │   ├── models.py
+    │   └── dependencies.py
+    └── tests/
+        ├── conftest.py
+        └── test_api.py
+
+Setting Up the Test Database
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+In ``conftest.py``, create fixtures for your test database:
+
+.. code-block:: python
+
+    import pytest
+    from sqlalchemy import create_engine
+    from sqlalchemy.orm import sessionmaker
+    from fastapi.testclient import TestClient
+    from app.main import app
```

Comment:
When testing it's best to use the create_app pattern so you create a new instance of app for each test